crowbar_framework/app/models/api/crowbar.rb

```diff
@@ -14,7 +14,90 @@
 # limitations under the License.
 #
 
+require "open3"
+
 module Api
-  class Crowbar
+  class Crowbar < Tableless
+    attr_reader :version
+
+    def initialize
+      @version = ENV["CROWBAR_VERSION"]
+    end
+
+    def status
+      {
+        version: @version
+      }
+    end
+
+    def upgrade
+      status.merge!(
+        upgrade: {
+          upgrading: upgrading?,
+          success: success?,
+          failed: failed?
+        }
+      )
+    end
+
+    def upgrade!
+      if upgrading?
+        errors.add(:upgrade, I18n.t("api.crowbar.upgrade_ongoing"))
+        return false
+      end
+
+      upgrade_script = Rails.root.join("..", "bin", "upgrade_admin_server.sh")
+
+      if upgrade_script.exist?
+        pid = spawn("sudo #{upgrade_script}")
+        Process.detach(pid)
+        Rails.logger.info("#{upgrade_script} executed with pid: #{pid}")
+
+        true
+      else
+        msg = "Could not find #{upgrade_script}"
+        Rails.logger.error(msg)
+        errors.add(:upgrade, msg)
+
+        false
+      end
+    end
+
+    def maintenance_updates_missing?
```

Comment:
> I think you can add some logging - or maybe different return values for maintenance_updates_installed?, we will need to tell user about specific failure

Added some specific logging, but for the business logic I think for those two cases it doesn't really make a difference for the user which updates are missing. both need to be installed anyways.